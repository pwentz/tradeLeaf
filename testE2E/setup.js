import React, { Component } from 'react';
const { View, StyleSheet } = require('react-native');

import { Provider } from 'react-redux';

import MockApi from './mocks/MockApi';
const { setupApp } = require('../setupApp');

__DEV__ = true;

jest.mock('Linking', () => {
  return {
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    openURL: jest.genMockFn().mockReturnValue(Promise.resolve()),
    canOpenURL: jest.genMockFn().mockReturnValue(Promise.resolve()),
    getInitialURL: jest.genMockFn().mockReturnValue(Promise.resolve()),
  };
});

jest.mock('ScrollView', () => jest.genMockFromModule('ScrollView'));

export const renderer = require('react-test-renderer');

const AccountRequirements = require('../src/components/register/AccountRequirements').default;
const App = require('../src/components/app/App').default;
const Avatar = require('../src/components/common/Avatar').default;
const Card = require('../src/components/matchboard/Card').default;
const LoginForm = require('../src/components/login/LoginForm').default;
const RegisterForm = require('../src/components/register/RegisterForm').default;
const TabBar = require('../src/components/app/TabBar').default;
const ChatPreview = require('../src/components/chat/ChatPreview').default;

export const components = {
  LoginForm,
  RegisterForm,
  Card,
  AccountRequirements,
  Avatar,
  TabBar,
  ChatPreview,
};

class TestRouteObserver {}

export function setupTestApp() {
  const api = new MockApi();

  const shouldLog = false;

  const { actions, store } = setupApp(api, shouldLog);

  const styles = StyleSheet.create({
    container: {
      flex: 1,
      justifyContent: 'center',
      alignItems: 'stretch',
      backgroundColor: 'white',
    },
  });

  const testRouteObserver = new TestRouteObserver();

  class TradeLeafApp extends Component {
    render() {
      return (
        <View style={styles.container}>
          <Provider store={store}>
            <App screenProps={{ actions, testRouteObserver }} />
          </Provider>
        </View>
      );
    }
  }

  return {
    TradeLeafApp,
    api,
    actions,
    dispatch: store.dispatch,
    testRouteObserver,
  };
}
