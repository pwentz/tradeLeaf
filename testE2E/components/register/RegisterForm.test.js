import React from 'react';

import { components, renderer } from '../../setup';

const noop = () => {}

it('<RegisterForm />', () => {
  const RegisterForm = components.RegisterForm;
  const component = renderer.create(
    <RegisterForm
      onSubmitRegister={noop}
      backToLogin={noop}
    />
  );
  expect(component.toJSON()).toMatchSnapshot();
})
