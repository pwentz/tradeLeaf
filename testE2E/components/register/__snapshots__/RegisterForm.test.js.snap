// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegisterForm /> 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "alignSelf": "stretch",
      "backgroundColor": "#F9F9F9",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "height": 35,
      }
    }
  />
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "marginBottom": 25,
        "marginRight": 25,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            Object {
              "borderColor": "#B8B8B8",
              "borderRadius": 25,
              "borderWidth": 1,
            },
          ],
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#B8B8B8",
            "fontSize": 16,
            "marginBottom": 4,
            "marginLeft": 10,
            "marginRight": 10,
            "marginTop": 4,
            "textAlign": "center",
          }
        }
      >
        log in
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <Image
      source={1}
    />
    <View
      style={
        Object {
          "height": 20,
        }
      }
    />
  </View>
</View>
`;
